### ISSUE TITLE
Twitter::Autolink breaks in jruby when text contains Chinese or Japanese

### ISSUE THREAD

**Issue by larryzhao** [twitter-archive/twitter-text-rb#70]
I just has this test:

with `jruby-1.6.7.2 1.9 mode`:

```
$ jruby -S test.rb
=== original ===
@baozi hi you are a dog #dog
@baozi 嗨你是一条狗诶 #狗狗
@baozi ねえ、あなたは犬だ #犬
=== autolinked ===
<a class="tweet-url username" href="https://twitter.com/baozi" rel="nofollow">@baozi</a> hi you are a dog <a class="tweet-url hashtag" href="https://twitter.com/#!/search?q=%23dog" rel="nofollow" title="#dog">#dog</a>
@baozi 嗨你是一条狗诶 #狗<a class="tweet-url username" href="https://twitter.com/baozi" rel="nofollow">狗baozi</a><a class="tweet-url hashtag" href="https://twitter.com/#!/search?q=%23狗狗" rel="nofollow" title="#狗狗">@狗狗</a>@baozi 嗨你是一条狗诶 #狗狗
@baozi ねえ、あなたは犬だ #<a class="tweet-url username" href="https://twitter.com/baozi" rel="nofollow">犬baozi</a><a class="tweet-url hashtag" href="https://twitter.com/#!/search?q=%23犬" rel="nofollow" title="#犬">@犬</a>@baozi ねえ、あなたは犬だ #犬
```

The autolink breaks when test contains Chinese/Japanese. But this only happens with jruby, if I use MRI ruby, everything is fine.

The code I use to test is here: https://gist.github.com/3045668

**Comment by larryzhao** [comment]
After a few debug, I see that it's because MatchData.begin(n) always returns 0 on jruby 1.9 mode when there's chinese/japanese in the text, see https://gist.github.com/3052569

Now sure if I've missed some jruby settings. I've also opened a SO question for it: http://stackoverflow.com/questions/11341449/matchdata-beginn-always-returns-0-on-jruby

**Comment by larryzhao** [comment]
Just had a talk in jruby irc, and found out that it's fixed in 1.7.0, you could see here: https://gist.github.com/3052627 this one is tested with 1.7.0.preview2. I've also tested with 1.7.0.preview1, also working correctly.
